components/renderers/controls/array/array-renderer.ts

```diff
@@ -2,80 +2,105 @@
         }
 
         defaultGridOptions.onRegisterApi = (gridApi) => {
-            if  (paginationEnabled) {
+            if (externalPaginationEnabled) {
                 gridApi.pagination.on.paginationChanged(this.scope, (newPage, pageSize) => {
                     defaultGridOptions.paginationCurrentPage = newPage;
                     defaultGridOptions.paginationPageSize = pageSize;
                     dataProvider.setPageSize(pageSize);
-                    dataProvider.fetchPage(newPage, pageSize).then(newData => {
-                        defaultGridOptions.data = newData;
+                    dataProvider.fetchPage(newPage).then(newData => {
+                        existingOptions.data = newData;
                     });
                 });
             }
-            gridApi.edit.on.afterCellEdit(this.scope, (rowEntity, colDef: uiGrid.IColumnDef, newValue, oldValue) => {
-                var value = newValue;
-
-                // TODO: newValue is a string?
-                if (colDef.type) {
-                    if (colDef.type == "number" || colDef.type == "integer") {
-                        value = Number(newValue);
-                    } else if (colDef.type == "boolean") {
-                        value = Boolean(newValue);
-                    }
-                }
-
-                rowEntity[colDef.field] = value;
+            if (externalFilteringEnabled) {
+                gridApi.core.on.filterChanged(this.scope, () => {
+                    var columns = gridApi.grid.columns;
+                    var terms = columns.reduce((acc, column) => {
+                        var value: any = column.filters[0].term;
+                        acc[column.field] = ArrayRenderer.convertColumnValue(column.colDef, value);
+                        return acc;
+                    }, {});
+                    dataProvider.filter(terms).then(newData => {
+                        existingOptions.data = newData;
+                    })
+                });
+            }
+            gridApi.edit.on.afterCellEdit(this.scope, (rowEntity, colDef:uiGrid.IColumnDef, newValue, oldValue) => {
+                rowEntity[colDef.field] = ArrayRenderer.convertColumnValue(colDef, newValue);
                 validationService.validate(rowEntity, schema['items']);
                 // TODO: use constant
                 gridApi.core.notifyDataChange("column");
                 this.scope.$apply();
             });
-
         };
 
-        return defaultGridOptions;
+        return this.mergeOptions(existingOptions, defaultGridOptions);
     }
 
-    private maxSize = 99;
-    priority = 2;
+    private static convertColumnValue(colDef: uiGrid.IColumnDef, value: string): any {
+        if (colDef.type) {
+            if (colDef.type == "number" || colDef.type == "integer") {
+               return Number(value);
+            } else if (colDef.type == "boolean") {
+                return Boolean(value);
+            }
```

Comment:
I think we don't have to handle `Date`s here, because JSON schema doesn't have a `Date` type, but a string with `format` being set to `date-time`.